pifpaf/drivers/__init__.py

```diff
@@ -63,33 +63,21 @@ def putenv(self, key, value, raw=False):
         self.env[key] = value
         return self.useFixture(fixtures.EnvironmentVariable(key, value))
 
-    def _kill(self, pid, sig=signal.SIGTERM):
-        os.kill(pid, sig)
-
-        # Wait 10 seconds max
+    def _kill(self, process):
+        pgrp = os.getpgid(process.pid)
+        process.terminate()
         try:
-            self._wait(pid)
+            tenacity.retry(
```

Comment:
you could define a `Retrying` object once and for all on the class and then call it here.
It'd be cheaper. :) not a big deal though:

```
_WAIT_FOR_POLL_NONE = tenacity.Retrying(wait=tenacity.wait_fixed(1),
                 stop=tenacity.stop_after_attempt(10),
                 retry=tenacity.retry_if_result(lambda ret: ret is None))

[...]

try:
    self._WAIT_FOR_POLL_NONE(process.poll)
except tenacity.RetryError:
    [...]
```